File: app/src/main/java/com/example/ui_exmaple/ui/home/HomeFragment.java
package com.example.ui_exmaple.ui.home;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.ui_exmaple.PermissionProxy;
import com.example.ui_exmaple.R;
import com.example.ui_exmaple.widget.SegmentProgressBar;
import com.yanzhenjie.permission.AndPermission;
import com.yanzhenjie.permission.runtime.Permission;

import org.jetbrains.annotations.NotNull;

import timber.log.Timber;

public class HomeFragment extends Fragment {
    private SegmentProgressBar mTextHome = null;
    private View mRoot;

    @Nullable
    @org.jetbrains.annotations.Nullable
    @Override
    public View onCreateView(@NonNull @NotNull LayoutInflater inflater, @Nullable @org.jetbrains.annotations.Nullable ViewGroup container, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        mRoot = inflater.inflate(R.layout.fragment_home, container, false);
        mTextHome = mRoot.findViewById(R.id.text_home);
        return mRoot;
    }

    @Override
    public void onResume() {
        super.onResume();

        mTextHome.start();

        String keySource = HomeFragment.class.getSimpleName();
//        PermissionProxy proxy = new PermissionProxy.Builder()
//                .setKeySource(keySource)
//                .setRawRationale((context, data, executor) -> {
//                    Timber.tag(keySource).i("== RawRationale ==");
//                })
//                .setRawGranted(data -> {
//                    Timber.tag(keySource).i("== RawGranted ==");
//                })
//                .setRawDenied(permissions -> {
//                    Timber.tag(keySource).i("== RawDenied ==");
//                })
//                .create();
//
//        AndPermission.with(this)
//                .runtime()
//                .permission(Permission.CAMERA)
//                .rationale(proxy.getRationale())
//                .onGranted(proxy.getGranted())
//                .onDenied(proxy.getDenied())
//                .start();
    }

    @Override
    public void onPause() {
        mTextHome.stop();
        super.onPause();
    }
}

File: app/src/main/java/com/example/ui_exmaple/ui/home/HomeFragment.kt
package com.example.ui_exmaple.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.ui_exmaple.R
import com.example.ui_exmaple.manager.heartBeat
import com.example.ui_exmaple.view.SegmentProgressBar

class HomeFragment : Fragment() {

    private lateinit var homeViewModel: HomeViewModel
    private var mTextHome:SegmentProgressBar? = null
    private var mRoot: View? = null
    private var mAutoRunnable:Runnable? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        homeViewModel =
            ViewModelProvider(this).get(HomeViewModel::class.java)
        mRoot = inflater.inflate(R.layout.fragment_home, container, false)
        mTextHome = mRoot?.findViewById(R.id.text_home)
        mAutoRunnable = Runnable {
            mTextHome?.setProgress(heartBeat.getInstance().progress)
            mTextHome?.postDelayed(mAutoRunnable,1000)
        }
        return mRoot
    }

    override fun onResume() {
        super.onResume()
        mTextHome?.setProgress(heartBeat.getInstance().progress)
        onStartRecycle()
    }

    override fun onPause() {
        super.onPause()
        stopRecycle()
    }

    private fun onStartRecycle(){
        mTextHome?.post(mAutoRunnable)
    }

    private fun stopRecycle(){
        mTextHome?.removeCallbacks(mAutoRunnable)
    }
}